src/PaymentGateways/Stripe/Controllers/NewStripeAccountOnBoardingController.php

```diff
@@ -38,6 +42,15 @@ public function __invoke()
             return;
         }
 
+        $isDonationFormPage = isset($_GET['give_tab']) && $_GET['give_tab'] !== 'stripe_form_account_options';
+        $isSettingPage = isset($_GET['tab'], $_GET['tab'])
+                         && Give_Admin_Settings::is_setting_page('gateways', 'stripe-settings');
+
+        // Exit if admin is not redirect to the GiveWP settings page or donation form page.
+        if (! $isDonationFormPage && ! $isSettingPage) {
+            return;
+        }
+
```

Comment:
Wouldn't this need to be _or_ not _and_?

```suggestion
        if (!$isDonationFormPage || !$isSettingPage) {
            return;
        }

```

Reply:
@jonwaldstein, the Stripe Connect flow should ideally redirect back to either the v2 donation form edit page or the Stripe global settings page upon completion. If the redirection target is neither, then the process should terminate gracefully.

Reply:
Oh I see, I think I just got confused reading it. Would be better to make a `$this->isPageValid()` or something to its more clear

Reply:
@jonwaldstein i added a function `$this->canProcessRequestOnCurrentPage`